### DISCUSSION TITLE
autoReconnect after calling a2dp_sink.end and a2dp_sink.start

### DISCUSSION THREAD

**Discussion by SimosMCmuffin** [pschatzmann/ESP32-A2DP#313] [Q&A] [answered]
Hi,

Been building a little portable audio development board and I'm wondering if it's possible to reconnect to the last connected device automatically after shutting down the a2dp_sink with a an .end() and restarting with .start() again later?

Use case:
The board has a 3.5mm audio jack as another source and there is currently a way to toggle the audio input between either the bluetooth or that 3.5mm jack. If the jack is used, I want to shutdown the bluetooth radio to conserve battery (current consumption ~100mA via the 3.5mm jack and about ~175mA via the BT playback) and then re-enable it after the source is toggled back to the bluetooth and then preferably automatically reconnect to the last connected device after that.

Currently the auto-reconnect works if the a2dp_sink keeps running in the background and the phone's bluetooth is switched off and back on again, no problem. But if the service is toggled off with .end() and then restarted with .start(), then it's required to manually reconnect from the phone to the a2dp_sink. 

Not sure if calling .end(), the last connected device is saved or discarded, or when calling .start() the last connected device is discarded.

Otherwise a very easy to use a2dp bluetooth library +1

**Comment by pschatzmann** [comment, marked as answer]
If you double check in the source code, calling end() is clearning the last connected device. Feel free to create your own subclass and overwrite the end() method as described  in the [design overview](https://github.com/pschatzmann/ESP32-A2DP/wiki/Design-Overview)

**Comment by SimosMCmuffin** [comment]
Just to give the answer to anybody else who might wonder about this later.

In "BluetoothA2DPCommon.cpp"
```
void BluetoothA2DPCommon::end(bool release_memory) {
    // reconnect should not work after end
    is_start_disabled = false;
    clean_last_connection();  //<<<--- the last connected device gets cleaned by this command when .end() is called
```
